engine/src/main/java/org/terasology/logback/RegexFilterAction.java

```diff
@@ -0,0 +1,180 @@
+ *     {@code /given scan \w+ are empty/}.</dd>
+ * </dl>
+ * <p>
+ * <b>Example:</b>
+ * <p><code>
+ *     &lt;requireRegex message="module:.*-SNAPSHOT" /&gt;
+ * </code>
+ * <dl>
+ *     <dt>{@code requireRegex}</dt>
+ *     <dd>The tag starts with {@code "require"}, meaning events <em>must</em> match the pattern.</dd>
+ *
+ *     <dt>{@code message}</dt>
+ *     <dd>Match the {@linkplain ILoggingEvent#getFormattedMessage() formatted message} against
+ *     the regular expression {@linkplain java.util.regex.Pattern pattern}
+ *     {@code /module:.*-SNAPSHOT/}.</dd>
+ * </dl>
+ * No Groovy or Janino required.
+ * <p>
+ * This Action adds an {@link EvaluatorFilter} with a {@link RegexEvaluator} to
+ * the current element, such as an {@link ch.qos.logback.core.Appender Appender}.
+ * <p>
+ * The element must be on the top of the stack and must implement {@link FilterAttachable}.
+ * <p>
+ * <b>Implementation Note:</b> In logback 1.2, we can <em>almost</em> define an
+ * EvaluatorFilter using only logback-core, but core defines no LoggingEvent interface.
+ * We have to link against logback-classic for that.
+ *
+ * @param <E> the type of logging event the filter operates on
+ */
+public class RegexFilterAction<E extends ILoggingEvent> extends Action {
+    public static final String PREFIX_ATTRIBUTE = "prefix";
+    public static final String MESSAGE_ATTRIBUTE = "message";
+    public static final String DENY_ELEMENT = "deny";
+    public static final String REQUIRE_ELEMENT = "require";
+
+    protected final BiFunction<String, String, EventEvaluator<E>> evaluatorConstructor = RegexEvaluator::new;
+
+    @Override
+    public void begin(InterpretationContext ic, String elementName, Attributes attributes) throws ActionException {
+        if (peekFilterContainer(ic) == null) {
+            return;
+        }
+
+        var evaluator = evaluatorConstructor.apply(
+                findAndSubst(ic, attributes, PREFIX_ATTRIBUTE),
+                findAndSubst(ic, attributes, MESSAGE_ATTRIBUTE)
+        );
+
+        EvaluatorFilter<E> filter = new RegexFilter<>(
+                getContext(),
+                evaluator,
+                makeFilterName(ic, attributes));
+
+        if (elementName.startsWith(REQUIRE_ELEMENT)) {
+            filter.setOnMatch(FilterReply.NEUTRAL);
+            filter.setOnMismatch(FilterReply.DENY);
+        } else if (elementName.startsWith(DENY_ELEMENT)) {
+            filter.setOnMatch(FilterReply.DENY);
+            filter.setOnMismatch(FilterReply.NEUTRAL);
+        }
+
+        // Push the new filter to the stack, in case anyone wants to configure it further.
+        ic.pushObject(filter);
+    }
+
+    @Override
+    public void end(InterpretationContext ic, String name) throws ActionException {
+        Object top = ic.peekObject();
+        EvaluatorFilter<E> filter;
+
+        if (top instanceof EvaluatorFilter) {
+            //noinspection unchecked
```

Comment:
```suggestion
            // noinspection unchecked
```
I don't really understand what this comment means, can we maybe elaborate it a little bit?

Reply:
`noinspection` comments are for IntelliJ's inspections; it's warning about the cast from `Object` to `EvaluatorFilter<E>`, despite the branch being guarded by the `instanceof EvaluatorFilter` clause.

Reply:
```suggestion
            // ignore Java warning about cast from Object to EvaluatorFilter<E>
            //noinspection unchecked
```